### ISSUE TITLE
Implement SASL SCRAM SHA 256 PLUS (2nd part)

### ISSUE THREAD

**Issue by igrishaev** [igrishaev/pg2#22]
A followup of #21 

Links for the context:

https://github.com/ongres/scram
https://github.com/jorsol/pgjdbc/blob/2f7511504c8eff34e55ed4b72df369687a09e3a8/pgjdbc/src/main/java/org/postgresql/core/v3/ScramAuthenticator.java#L53

code:

```java
SSLSession session = ((SSLSocket) socket).getSession();
      try {
        Certificate[] certificates = session.getPeerCertificates();
        if (certificates != null && certificates.length > 0) {
          Certificate peerCert = certificates[0]; // First certificate is the peer's certificate
          if (peerCert instanceof X509Certificate) {
            X509Certificate cert = (X509Certificate) peerCert;
            return TlsServerEndpoint.getChannelBindingData(cert);
          }
        }

 public static byte @NotNull [] getChannelBindingData(final @NotNull X509Certificate serverCert)
      throws CertificateEncodingException {
    MessageDigest digestAlgorithm = getDigestAlgorithm(serverCert.getSigAlgName());
    if (digestAlgorithm == null) {
      return new byte[0];
    }
    return digestAlgorithm.digest(serverCert.getEncoded());
  }

```

**Comment by igrishaev** [comment]
Done, closing. Now we have -PLUS version of SCRUM working. cc @thenonameguy fyi

**Comment by igrishaev** [comment]
available in `[0.1.20]` on clojars

**Comment by Neustradamus** [comment]
@igrishaev: Good job for your commits!

Merged commits:
- https://github.com/igrishaev/pg2/commit/2bd89708355511b1584ba11e7bddb114ef51189f
- https://github.com/igrishaev/pg2/commit/8e2c6e8662b12a782701b787b1745d28a94603a4
- https://github.com/igrishaev/pg2/commit/ce4c38ce60fb5af5b78ca9f9b947914e48c9ac65

Linked to:
- https://github.com/scram-sasl/info/issues/1
- https://github.com/igrishaev/pg2/issues/21
- https://github.com/igrishaev/pg2/issues/22
